src/switchroot/ostree-prepare-root.c

```diff
@@ -506,29 +513,14 @@ main (int argc, char *argv[])
       else
         {
           int errsv = errno;
-          const char *errmsg;
-          switch (errsv)
-            {
-            case ENOVERITY:
-              errmsg = "fsverity not enabled on composefs image";
-              break;
-            case EWRONGVERITY:
-              errmsg = "Wrong fsverity digest in composefs image";
-              break;
-            case ENOSIGNATURE:
-              errmsg = "Missing signature for fsverity in composefs image";
-              break;
-            default:
-              errmsg = strerror (errno);
-              break;
-            }
-          if (composefs_config->enabled == OT_TRISTATE_MAYBE)
+          g_assert (composefs_config->enabled != OT_TRISTATE_NO);
+          if (composefs_config->enabled == OT_TRISTATE_MAYBE && errsv == ENOENT)
```

Comment:
Are we sure we want to errx in the ENOVERITY, EWRONGVERITY, ENOSIGNATURE cases when composefs is set to maybe?

Reply:
Also, why not have ENOENT in the switch statement with all the other error messages?

```
    case ENOENT:
      return "No image present";
```

Reply:
> Are we sure we want to errx in the ENOVERITY, EWRONGVERITY, ENOSIGNATURE cases when composefs is set to maybe?

[The commit message discusses this](
https://github.com/ostreedev/ostree/pull/2994/commits/38880bff01a4ed858b8b3d53aa00057b27d5bfc9); I'm open to debate but the rationale is there.

Reply:
> Also, why not have ENOENT in the switch statement with all the other error messages?

Two reasons:

- `ENOENT` is an "expected error", and we handle it specially anyways
- "No such file or directory" is already an OK explanation of the error

Ah yes and now that I look, the other errors don't actually exist, see:

```
#define ENOVERITY ENOTTY
#define EWRONGVERITY EILSEQ
#define ENOSIGNATURE EBADMSG
```

So the `strerror` for them would just be misleading and confusing.